src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -266,6 +262,10 @@ Char10: .
             skipCurrentTokenConsuming();
       } -> skip, mode(DEFAULT_MODE);
 
+fragment JavaLetter: [A-Za-z_$];
+fragment JavaLetterOrDigit: [0-9A-Za-z_$];
```

Comment:
We can't use `Character.isJavaIdentifierStart` and `Character.isJavaIdentifierPart` like we do in the Java grammar?
https://github.com/checkstyle/checkstyle/blob/25a37e5049b7816b34c552899841a978efc37a63/src/main/resources/com/puppycrawl/tools/checkstyle/grammars/java.g#L1878-L1935

Reply:
I can not see why we can not use it. But that would add semantic predicates and call to external function in the grammar which seems a bit unnecessary, don't you think ? 
Also, I feel the more explicit we can be with grammar definitions the better. The way I defined `JavaLetter` or `JavaLetterOrDigit` is the way how [JLS defines it too](https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-Identifier).

Reply:
lets not change this for now.

Reply:
@romani was there any benefit on why we did this in the Java grammar?